### PR TITLE
Adds Graphviz layer

### PR DESCRIPTION
**Pull request by luxbock**
- Uses a forked version of graphviz-dot-mode.el with a local install.
  Upstream PR: ppareit/graphviz-dot-mode#18

### CODE REVIEW

File: layers/+lang/graphviz/packages.el [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        "t" 'spacemacs/toggle-graphviz-live-reload
+        "c" 'compile
+        "p" 'graphviz-dot-preview
+        "," 'graphviz-dot-preview))))
+
+(defun graphviz/pre-init-popwin ()
+  (spacemacs|use-package-add-hook popwin
+    :post-config
+    (push '("*compilation*" :dedicated t :position bottom :stick t :noselect nil :height 0.4)
+          popwin:special-display-config)))
```
Comment (nixmaniack) [inline layers/+lang/graphviz/packages.el]: Spacemacs already takes care of [this](https://github.com/syl20bnr/spacemacs/blob/72e1413e22c193ffe0b8d69b1b153e4e0ec14cf5/layers/+distribution/spacemacs-base/packages.el#L739).

Comment (luxbock) [inline layers/+lang/graphviz/packages.el]: Ah I see, yeah I'll remove that then.

**Comment by robbyoconnor** [comment]
Nice job! :heart:

**Comment by syl20bnr** [comment]
Consider using a recipe location pointing on your GitHub fork instead of a local package, I'm not fond of importing 1k LoC temporarily ;-)
You can find an example of such a recipe here: https://github.com/syl20bnr/spacemacs/blob/develop/layers/+lang/php/packages.el#L21

**Comment by luxbock** [comment]
Should be fixed now.

**Comment by robbyoconnor** [comment]
@luxbock make sure that you keep your branch up to date and monitor upstream :)

**Comment by AdonRain** [comment]
awesome !

**Comment by peterhoeg** [comment]
Any chance you can change the :mode line to this instead:

```
    :mode (("\\.diag$"      . graphviz-dot-mode)
           ("\\.blockdiag$" . graphviz-dot-mode)
           ("\\.nwdiag$"    . graphviz-dot-mode)
           ("\\.rackdiag$"  . graphviz-dot-mode)
           ("\\.dot$"       . graphviz-dot-mode)
           ("\\.gv"         . graphviz-dot-mode))
```

Also, you can get rid of ```:defer t``` as it is implied when using :mode or :commands.

**Comment by robbyoconnor** [comment]
> Also, you can get rid of :defer t as it is implied when using :mode or :commands.

Huh, no, leave it.

**Comment by peterhoeg** [comment]
Straight from the docs of use-package:
```
If you aren't using :commands, :bind, :bind*, :bind-keymap, :bind-keymap*, :mode, or :interpreter (all of which imply :defer; see the docstring for use-package for a brief description of each), you can still defer loading with the :defer keyword:
```

**Comment by robbyoconnor** [comment]
> Straight from the docs of use-package:
> 
> If you aren't using :commands, :bind, :bind, :bind-keymap, :bind-keymap, :mode, or :interpreter (all of which imply :defer; see the docstring for use-package for a brief description of each), you can still defer loading with the :defer keyword:

Heh in that case, ignore what I just said and follow @peterhoeg's advice

**Comment by luxbock** [comment]
Should be fixed by

**Comment by a13ph** [comment]
Some inspirations for what can be done with graphwiz and org-mode <3 http://pages.sachachua.com/evil-plans/ 
(And if somebody makes this very easy to do, I'd be very happy)

**Comment by syl20bnr** [comment]
Thank you :+1: 
I made some changes to the key bindings and added support for smarparens, related commit: https://github.com/syl20bnr/spacemacs/commit/130cd48
Cherry-picked into develop branch, you can safely delete your branch.

**Pull request closed without merging**